project_scrum/project_scrum.py

```diff
@@ -0,0 +1,500 @@
+                             string='State', required=False)
+    company_id = fields.Many2one(related='project_id.analytic_account_id.company_id')
+
+    # Compute: effective_hours, total_hours, progress
+    @api.one
+    def _hours_get(self):
+        res = {}
+        effective_hours = 0.0
+        planned_hours = 0.0
+        for task in self.task_ids:
+            effective_hours += task.effective_hours or 0.0
+            planned_hours += task.planned_hours or 0.0
+        self.effective_hours = effective_hours
+        # self.planned_hours = planned_hours
+        return True
+
+    @api.onchange('project_id')
+    def onchange_project_id(self):
+        if self.project_id and self.project_id.manhours:
+            self.planned_hours = self.project_id.manhours
+        else:
+            self.planned_hours = 0.0
+
+    @api.onchange('date_start')
+    def onchange_date_start(self):
+        if self.date_start:
+            if self.project_id:
+                self.date_stop = fields.Date.from_string(self.date_start) +\
+                 timedelta(days=self.project_id.default_sprintduration)
+        else:
+            pass
+
+class project_user_stories(models.Model):
+    _name = 'project.scrum.us'
+    _description = 'Project Scrum Use Stories'
+    _order = 'reference'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    
+    @api.model
+    def create(self, vals):
+        vals['reference'] = self.env['ir.sequence'].get('user.story')
+        return super(project_user_stories, self).create(vals)
+     
+    name = fields.Char(string='User Story', required=True)
+    color = fields.Integer(related='project_id.color')
+    description = fields.Html(string='Description')
+    description_short = fields.Text(compute='_conv_html2text', store=True)
+    actor_ids = fields.Many2many(comodel_name='project.scrum.actors', string='Actor')
+    project_id = fields.Many2one(comodel_name='project.project', string='Project', ondelete='set null',
+        select=True, track_visibility='onchange', change_default=True)
+    sprint_ids = fields.Many2many(comodel_name='project.scrum.sprint', string='Sprint', store=True)
+    task_ids = fields.One2many(comodel_name='project.task', inverse_name='us_id')
+    task_test_ids = fields.One2many(comodel_name='project.scrum.test', inverse_name='user_story_id_test')
+    task_count = fields.Integer(compute='_task_count', store=True)
+    test_ids = fields.One2many(comodel_name='project.scrum.test', inverse_name='user_story_id_test')
+    test_count = fields.Integer(compute='_test_count', store=True)
+    sequence = fields.Integer('Sequence')
+    company_id = fields.Many2one(related='project_id.analytic_account_id.company_id')
+    moscow = fields.Selection([
+            (1, 'must'), 
+            (2, 'should'), 
+            (3, 'could'), 
+            (4, 'wont'), 
+            ('not_set', 'Not Set'), 
+            ], string='Moscow')
+    value = fields.Selection([
+             (0, '0'),
```

Comment:
wouldnot it be easier  to store these values in a model?

Reply:
that or a function that generates the values, also more customization friendly